dotnet/src/Functions/Functions.OpenAPI/RestApiOperationRunner.cs

```diff
@@ -171,7 +173,34 @@ internal sealed class RestApiOperationRunner
 
         using var responseMessage = await this._httpClient.SendWithSuccessCheckAsync(requestMessage, cancellationToken).ConfigureAwait(false);
 
-        return await SerializeResponseContentAsync(responseMessage.Content).ConfigureAwait(false);
+        var response = await SerializeResponseContentAsync(responseMessage.Content).ConfigureAwait(false);
+
+        if (responses is not null)
+        {
+            var statusCodeKey = $"{(int)responseMessage.StatusCode}";
+
+            // Exact Match
+            var matchingResponse = responses.FirstOrDefault(r => r.Key == statusCodeKey).Value;
+
+            // Wildcard match e.g. 2XX
+            if (matchingResponse is null)
+            {
+                matchingResponse = responses.FirstOrDefault(r => r.Key.EndsWith("X", StringComparison.InvariantCultureIgnoreCase) && statusCodeKey.StartsWith(r.Key.Substring(0, 1), StringComparison.InvariantCultureIgnoreCase)).Value;
```

Comment:
This only matches first and last characters. Can we convert the key to a regex and use that to match?

Reply:
The intent is to match with the first digit of the response code and XX. Could change to compare with `$"{statusCodeKey.Substring(0,1)}XX"`. If the response code is 403 and 403 is not defined we should not match with any other 4** code other than the literal string "4XX".